Enemy movement

vincetiu8 opened the pull request:
Added more interesting enemy movement. Still need to figure out how to fix tower aiming though...

vincetiu8 committed:
Implement more interesting enemy movement

This makes the enemy movement less boring by moving them on both axes.

vincetiu8 committed:
Fixed enemy aiming slightly

Now the towers are sometimes able to hit the enemies. It's not a big, its a feature! Just kidding, we still need to fix this.

vincetiu8 committed:
Give towers aimbot

Tower bullets now track down enemies. At high speeds, they should still look like bullets. At low speeds, we can use this to create tracking missiles or whatever else we need.

Graphcalibur reviewed (changes requested):
Looks really good but there's a weird animation bug if you put towers near the starting point: https://youtu.be/T7xuY2psrDc

vincetiu8 committed:
Fix bugs

This fixes the bug where the bullets circle around where the enemy they were tracking died and also adds a lifetime to the bullets.

vincetiu8 committed:
Merge branch 'master' into enemy-movement
